src/main/java/com/notably/logic/commands/NewCommand.java

```diff
@@ -15,9 +13,9 @@
 public class NewCommand extends Command {
     public static final String COMMAND_WORD = "new";
     private final Block toAdd;
-    private Path path;
+    private AbsolutePath path;
```

Comment:
Hmmm, although this `path` here is used in the `execute` method to check for the existence of a path, I actually don't think it's very intuitive for the creator of the `NewCommand` to need to supply a `path`. At the end of the day, we're always adding the new `Block` to the currently open path. That means a single `Block toAdd` field should already be able to supply all the needed information, don't you think?

Reply:
Yes. That would mean `hasBlock` have to take in a `Block` instead of `Path` object. But can the current implementation of taking in a `Path` be more useful for future upgrade (e.g NewCommand that supports nested creation) ?

Reply:
> Yes. That would mean `hasBlock` have to take in a `Block` instead of `Path` object. But can the current implementation of taking in a `Path` be more useful for future upgrade (e.g NewCommand that supports nested creation) ?

Actually, I don’t think so. You can just try to add the `Block` in, and catch the `DuplicateBlockException` instance. Then, you rethrow your current exception.